src/Collection/Iterator/UniqueIterator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Collection\Iterator;
+
+class UniqueIterator extends FilterIterator
+{
+    protected array $_exists = [];
+
+    /**
+     * @param \Traversable|array $items The items to be filtered.
+     * @param string|null $key An object property or array key that will be used to determine uniqueness.
```

Comment:
How about callable support?

Reply:
With a callable it would be effectively same as filter()/reject() isn't it? I think unique() meant for simpler cases when one doesn't want to bother writing their own callable.